test/metric_learn_test.py

```diff
@@ -76,6 +76,111 @@ def test_singular_returns_pseudo_inverse(self):
                     pseudo_inverse)
 
 
+class TestSCML():
+  def test_iris(self):
+    X, y = load_iris(return_X_y=True)
+    scml = SCML_Supervised(n_basis=80, k_genuine=8, k_impostor=11,
+                           random_state=42)
+    scml.fit(X, y)
+    csep = class_separation(scml.transform(X), y)
+    assert csep == 0.24555420119538296
+
+  @pytest.mark.parametrize(('estimator', 'data', 'authorized_basis'),
+                           [(SCML, (np.ones((3, 3, 3)),), ['triplet_diffs']),
+                            (SCML_Supervised, (np.array([[0, 0], [0, 1],
+                                                         [2, 0], [2, 1]]),
+                                               np.array([1, 0, 1, 0])),
+                             ['triplet_diffs', 'lda'])])
+  def test_bad_basis(self, estimator, data, authorized_basis):
+    model = estimator(basis='bad_basis')
+    msg = ("`basis` must be one of the options '{}' or an array of shape "
+           "(n_basis, n_features).".format("', '".join(authorized_basis)))
```

Comment:
you could maybe just reuse the attribute `_authorized_basis` from `SCML` and `SCML_Supervised`?

Reply:
Great idea!